Sources/StreamChat/Controllers/DataController.swift

```diff
@@ -24,7 +24,14 @@ public class DataController: Controller {
     public internal(set) var state: State = .initialized {
         didSet {
             callback {
-                self.stateMulticastDelegate.invoke { $0.controller(self, didChangeState: self.state) }
+                // We only report state changes if not `.remoteDataFetched` because the data
+                // coming from the server will only be available once `FRC` reports changes,
+                // so when that happens, only then we report `didChangeState`. This is required
+                // mostly because of background mapping.
+                // More context here: https://github.com/GetStream/stream-chat-swift/pull/3039
+                if self.state != .remoteDataFetched {
```

Comment:
Also here, I would prefer if we limit this state check to BM enabled only, wdyt?

Reply:
We agreed on doing this change only with BG Mapping enabled 👍